### PR TITLE
added category entity and controller for it

### PR DESCRIPTION
**Pull request by DZarudnyi**

### CODE REVIEW

**Comment by DZarudnyi** [comment]
Keep getting LaziInitializationException, can't really understand why

File: src/main/java/com/example/books/service/category/CategoryService.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    List<CategoryDto> findAll(Pageable pageable);
+
+    CategoryDto getById(Long id);
+
+    CategoryDto save(CategoryDto categoryDto);
+
+    CategoryDto update(Long id, CategoryDto categoryDto);
+
+    void deleteById(Long id);
+
```
Comment (okuzan) [inline src/main/java/com/example/books/service/category/CategoryService.java:17]:
```suggestion
```

File: src/main/resources/db/changelog/changes/10-create-books-categories-table.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            baseColumnNames: book_id
+            constraintName: FK_book_id
+            referencedTableName: books
+            referencedColumnNames: id
+        - addForeignKeyConstraint:
+            baseTableName: books_categories
+            baseColumnNames: category_id
+            constraintName: FK_category_id
+            referencedTableName: categories
+            referencedColumnNames: id
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/10-create-books-categories-table.yaml:34]:
```suggestion
            referencedColumnNames: id

```

**Review by okuzan — APPROVED** [review]
Do you still have that problem? Please start a thread in the chat instead of submitting your solution in such cases

**Review by okuzan — APPROVED** [review]

File: src/main/resources/db/changelog/changes/10-create-books-categories-table.yaml (hunk 2 above)
Comment (okuzan) [reply src/main/resources/db/changelog/changes/10-create-books-categories-table.yaml:34]: add empty line

File: src/main/java/com/example/books/service/category/CategoryService.java (hunk 1 above)
Comment (okuzan) [reply src/main/java/com/example/books/service/category/CategoryService.java:17]: remove empty line

**Review by olekskov — APPROVED** [review]

**Pull request merged**